src/test/java/odysseusAT/CharacterizationStepDefs.java

```diff
@@ -0,0 +1,383 @@
+package odysseusAT;
+
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.Selectors;
+import cucumber.api.PendingException;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static com.codeborne.selenide.Condition.text;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selectors.byText;
+import static com.codeborne.selenide.Selenide.*;
+
+public class CharacterizationStepDefs {
+
+    private String generatedString;
+    private String characterizationName;
+    private String generatedStringFeature;
+    private String featureName;
+
+
+    @Then("^can see Characterization page$")
+    public void canSeeCharaterizationPage() {
+        $(By.xpath("//*[@data-bind='text: title']")).shouldHave(text("Cohort Characterizations"));
+    }
+
+    @When("^click to New characterization button$")
+    public void clickToNewCharacterizationButton() {
+        $(By.xpath("//*[@class='characterizations-tabbed-grid__new-entity-btn btn btn-primary btn-sm']")).click();
+        $(By.xpath("//*[@data-bind='text: title']")).shouldHave(text("New Characterization"));
+
+    }
+
+    @When("^enter Characterization name and save it$")
+    public void enterCharacterizationNameAndSaveIt() {
+        generatedString = RandomStringUtils.randomAlphanumeric(10);
+        characterizationName = "Test_" + generatedString;
+        $(By.xpath("//*[@class='input-group']/input")).clear();
+        $(By.xpath("//*[@class='input-group']/input")).setValue(characterizationName);
+        $(By.xpath("//*[@class='fa fa-save']")).click();
+    }
+
+    @When("^return to Characterization table$")
+    public void returnToCharacterizationTable() {
+        $(By.xpath("//*[@class='input-group-btn']/button[2]")).click();
```

Comment:
How did you identify the button? This way doesn't emulate user's logic at all: where is selection by text / icon?

Reply:
changed to $$(By.xpath("//*[@class='btn btn-primary']")).get(0).click();